### ISSUE TITLE
Unable to switch between different locators

### ISSUE THREAD

**Issue by prashanth-sams** [SeleniumHQ/selenium-ide#150]
When I try to switch the locator from ID to Css/XPath, the target field fail to provide the option for it.

![locator](https://user-images.githubusercontent.com/2948696/38776221-94b74bda-40a4-11e8-9ac3-f301c68d43e1.png)

**Comment by corevo** [comment]
The markup?

**Comment by prashanth-sams** [comment]
@corevo Yes, the recorded script's source (html formatted)

**Comment by corevo** [comment]
Unfortunately we moved to a better way of saving the scripts, one that allows us to contain both test cases and suites together in one file.  
We did it so that if you edit a test case the change will propagate to all the places it is used.  
The [new format](https://www.json.org/) is less human-readable, perhaps in the future we can convert it to a more [readable format](http://yaml.org/) for display purposes only, and save in the old format.  
To sum up, the format is not very human readable, so at this time I can't see the upside of letting users see it.

**Comment by 98gmarquee** [comment]
Personally, I liked having the suites and scripts separate. It allowed for easy copying of other scripts and dropping them in the suite.

Also, the html format was easy to generate using shell script and a data input file. I have many instances where I use an internal jsp to pull data out of a db in a table format that I then copy into a txt file. Then I use  a shell script to parse the data into an output file I can copy and paste directly into a selenium ide html file.

I run the query once, visually inspect the data for accuracy, then copy/paste the data directly into the txt file. As an example it could be 20 columns and 100 rows, which is 2000 individual validations.

Doing that is far quicker than creating individual steps using the gui to validate each cell on each row.

**Comment by corevo** [comment]
The format is JSON which is highly acceptable serialization format, second only to XML.  
If you're creating them programmatically, it may even be easier than HTML, depending on the serializer you use.  
Keeping the suites and test cases separate is far harder now than before, the new extension format doesn't allow us to automatically load your test cases when you load a suite, you'll have to upload all dependent test cases one-by-one every time, because of this we have decided to unite the two, which also meant that HTML is no longer a good format for it, and we've moved to JSON, perhaps the better decision would've been YAML.

**Comment by 98gmarquee** [comment]
Is the id in the side file relevant? I don't know how I would generate that via a script.

This:
`{"id":"a573f914-7b45-4d45-b01e-624bcbc2a41a",`

**Comment by corevo** [comment]
As long as it's unique it can be whatever you want.

**Comment by corevo** [comment]
You can for example define an int starting at 0, and increment it every time.  
Just remember to convert it to a string.

**Comment by 98gmarquee** [comment]
This may be the wrong place to ask, but since the screenshot shows the value, why does this version put in the x,y coordinates for what is being clicked and what are the coordinates a reference for? From the 0,0 of the page or the element to where the input box/item is?

When doing what minimal testing I can do with the existing ide, I've been going back and deleting the coordinates and changing the command to "click" as I wasn't sure if they would impact a script ran on a computer with a different screen resolution as I have 3 or 4 resolutions I test in.

**Comment by corevo** [comment]
@98gmarquee The coordinates are relative to the element, 0,0 is the top left corner of the element.  
The element will always be clicked regardless of the coordinates, the coordinates is mostly used to test effects that relay on them, for example the material ripple effect.

**Comment by 98gmarquee** [comment]
Why does it default to "click at" vs "click"? Guessing that was on purpose, just curious as for the background as to why.

**Comment by corevo** [comment]
It is the most accurate event, not keeping the coordinates means we are losing record information.

Cheers,
Tom

**Comment by bcmoney** [comment]
@prashanth-sams 
In case it helps, I've thrown this converter together to go between SIDE and HTML formats: 
http://jsbin.com/zagemaw
I know there are reasons to move forward away from the HTML version, but I am hoping from the discussions and requests I've seen for 2-way import/export that it could still be considered?

**Comment by Shobhitfp** [comment]
How Can i run the json loop in Side format now?In the revious version it was easy to run in HTML format
Example
<tr>
	<td>forJson</td>
	<td>File Path</td>
	<td></td>
</tr>

<tr>
	<td>endForJson</td>
	<td></td>
	<td></td>
</tr>
</tbody></table>
</body>
</html>
